stdlib/random.gr

```diff
@@ -0,0 +1,128 @@
+import WasiRandom from "sys/random"
+import Result from "result"
+import Int32 from "int32"
+import Int64 from "int64"
+import WasmI32 from "runtime/unsafe/wasmi32"
+import WasmI64 from "runtime/unsafe/wasmi64"
+import Memory from "runtime/unsafe/memory"
+import DS from "runtime/dataStructures"
+
+record Random { seed: Int64, mut counter: Int64, mut initialized: Bool }
+
+let incCounter = random => {
+  random.counter = Int64.incr(random.counter)
+}
+
+// https://arxiv.org/pdf/2004.06278v3.pdf
+@disableGC
+let rec squares = (ctr: Int64, key: Int64) => {
+  // Need to implement with @disableGC because
+  // overflows are part of this algorithm
```

Comment:
All of the Int64 ops support overflows, though.

Reply:
I'm not sure why (possibly due to the equivalent of `WasmI32.wrapI64`?), but I was getting overflow errors when using the Grain functions.

Either way, there is an efficiency argument to be made as well, since our boxed numeric operations are not as efficient as working with the raw WASM types.

Reply:
Yeah, that's fine. Let's update the comment to just say that's why, though.